File: py/1045.py
def triangleTypes(a, b, c):
    if a >= (b + c):
        print('NAO FORMA TRIANGULO')
        return
        
    if a**2 == (b**2 + c**2):
        print('TRIANGULO RETANGULO')
        
    elif a**2 > (b**2 + c**2):
        print('TRIANGULO OBTUSANGULO')
        
    elif a**2 < (b**2 + c**2):
        print('TRIANGULO ACUTANGULO')
        
        
    if a == b == c:
        print('TRIANGULO EQUILATERO')
        
    elif a == b or a == c or b == c:
        print('TRIANGULO ISOSCELES')
        
triangle = list(map(float, input().split()))
h = max(triangle)
triangle.remove(h)
triangleTypes(h, triangle[0], triangle[1])

File: cpp/1045.cpp
#include <iostream>

using namespace std;

int main()
{
    double a, b, c;
    cin >> a >> b >> c;

    if (a < b) {
        a = a + b;
        b = a - b;
        a = a - b;
    }

    if (a < c) {
        a = a + c;
        c = a - c;
        a = a - c;
    }

    if (a >= b + c) {
        cout << "NAO FORMA TRIANGULO\n";

    } else {
        if (a * a == b * b + c * c) {
            cout << "TRIANGULO RETANGULO\n";
        } else if (a * a > b * b + c * c) {
            cout << "TRIANGULO OBTUSANGULO\n";
        } else {
            cout << "TRIANGULO ACUTANGULO\n";
        }

        if (a == b && b == c) {
            cout << "TRIANGULO EQUILATERO\n";
        } else if (a == b || a == c || b == c) {
            cout << "TRIANGULO ISOSCELES\n";
        }
    }

    return 0;
}